modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -1180,16 +1181,24 @@ private RuntimeException handleStartException(Throwable e) {
 
         LOG.debug(errMsg, e);
 
-        lifecycleManager.stopNode();
+        try {
+            lifecycleManager.stopNode().get();
+        } catch (Exception ex) {
+            IgniteException igniteException = new IgniteException(errMsg, ex);
+            igniteException.addSuppressed(e);
+
+            return igniteException;
+        }
 
         return new IgniteException(errMsg, e);
     }
 
     /**
      * Stops ignite node.
      */
-    public void stop() {
-        lifecycleManager.stopNode();
+    public void stop() throws ExecutionException, InterruptedException {
```

Comment:
I'd say that we should provide both stopAsync() an stop() in order to be consistent.